src/NHibernate/Criterion/PropertyExpression.cs

```diff
@@ -70,31 +70,25 @@ protected PropertyExpression(string lhsPropertyName, IProjection rhsProjection)
 
 		public override SqlString ToSqlString(ICriteria criteria, ICriteriaQuery criteriaQuery)
 		{
-			SqlString[] columnNames =
-				CriterionUtil.GetColumnNames(_lhsPropertyName, _lhsProjection, criteriaQuery, criteria);
-			SqlString[] otherColumnNames =
-				CriterionUtil.GetColumnNames(_rhsPropertyName, _rhsProjection, criteriaQuery, criteria);
+			var columnNames =
+				CriterionUtil.GetColumnNamesAsSqlStringParts(_lhsPropertyName, _lhsProjection, criteriaQuery, criteria);
+			var otherColumnNames =
+				CriterionUtil.GetColumnNamesAsSqlStringParts(_rhsPropertyName, _rhsProjection, criteriaQuery, criteria);
 
-			SqlStringBuilder sb = new SqlStringBuilder();
-			if (columnNames.Length > 1)
-			{
-				sb.Add(StringHelper.OpenParen);
-			}
-			bool first = true;
-			foreach (SqlString sqlString in SqlStringHelper.Add(columnNames, Op, otherColumnNames))
-			{
-				if (first == false)
-				{
-					sb.Add(" and ");
-				}
-				first = false;
-				sb.Add(sqlString);
-			}
+			if (columnNames.Length == 0)
```

Comment:
Why did you change order of checks? In most cases it's 1 so it was first check intentionally. And it should never be 0 - it's added just for consistency with the old logic.